[Zhang Zhiyao] iP

zzybluebell opened the pull request:
**### Instruction** 😃

> “Your mind is for having ideas, not holding them.” – David Allen (source)

zzyBluebell 's Duke frees your mind of having to remember things you need to do. It's,

text-based
easy to learn


*FAST SUPER FAST to use*

All you need to do is,

1.   download GUI version V0.2.
2.   Open terminal.
3.   Use <cd> command to navigate to the directory of jar file.
4.   Enter java -jar duke.jar (Prerequisites: jdk installed).
5.   add your ### tasks.
6.   let it manage your tasks for you
7.   And it is FREE!

Features:

[ ]  Managing `Todo` Tasks
[ ]  Managing `Event` Tasks
[ ]  Managing `Deadline` Tasks
[ ]  `Find` specific Tasks


Examples:
-  todo read book 

-  event meeting /at 2021-08-23 17:00
-  deadline return book /by 2021-08-20 12:00
-  find book

If you Java programmer, you can use it to practice Java too. Here's the Launcher class and main method:
``` java
/**
 * A launcher class to workaround classpath issues.
 */
public class Launcher {

    /**
     * The main method for launch the Main class
     * 
     * @param args
     */
    public static void main(String[] args) {
        launch(Main.class, args);
    }
}
```
 [Project details can be viewed here](https://github.com/zzybluebell/ip)

Commit:
Add Gradle support

zzybluebell committed:
The level one. Greet, Echo, Exit implementation

zzybluebell committed:
The level 2  Add, List implementation

zzybluebell committed:
Level 3 Mark as Done implementation

zzybluebell committed:
Level 3 modification

zzybluebell committed:
Level 4. ToDos, Events, Deadlines implementation

zzybluebell committed:
A-TextUiTesting t

zzybluebell committed:
Level 4. ToDos, Events, Deadlines modification

zzybluebell committed:
Level 5. Handle Errors implementation

zzybluebell committed:
Level 5. Handle Errors implementation and modification

zzybluebell committed:
Level 6. Delete implementation

zzybluebell committed:
A-TextUiTesting Test using the I/O redirection technique

zzybluebell committed:
A-Enums Use Enumerations implementation

zzybluebell committed:
A-TextUiTesting Test using the I/O redirection technique modication

zzybluebell committed:
Level 6 modification

zzybluebell committed:
Branch level7 readData

zzybluebell committed:
Branch level 7

zzybluebell committed:
Merge branch 'branch-Level-7'

zzybluebell committed:
level 8  Date time

zzybluebell committed:
Merge branch 'branch-Level-8'

zzybluebell committed:
More OOP version 1

zzybluebell committed:
More OOP

zzybluebell committed:
Merge branch 'MoreOOP'

zzybluebell committed:
A-packages

zzybluebell committed:
Merge branch 'A-Packages'

zzybluebell committed:
JUnit with 2 test method 2 test class

zzybluebell committed:
Merge branch 'A-JUnit'

zzybluebell committed:
A-JavaDoc

zzybluebell committed:
Merge branch 'A-Jar'

zzybluebell committed:
Level-9 find

zzybluebell committed:
Merge branch 'Level-9'

ykwei7 commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+package Duke;
```
```suggestion
package duke;
```

mingyi456 replied:
I believe packages should be named with all lowercase or something, so `duke` and not `Duke`?

ykwei7 commented on src/main/java/Duke/Exceptions/DeleteWrongIndexException.java:
```diff
@@ -0,0 +1,14 @@
+package Duke.Exceptions;
```
Might be easier to put all the DukeExceptions in a single Java file, and extend by doing public static class for sub-class exceptions

mingyi456 replied:
Also, just like your `Duke` package, your `Exceptions` package should be named in lower case, i.e. `exceptions`

ykwei7 reviewed (commented):
Looks good overall. 👍

ramapriyan912001 commented on src/main/java/Duke/Tasks/Find.java:
```diff
@@ -0,0 +1,32 @@
+public class Find extends Task {
+
+    protected String description;
+    /**
```
Perhaps it would be better for the readability of the code if you left a line space between the end of the field declarations and the constructor?

ramapriyan912001 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,10 @@
+todo eat
+todo cs2103 tut
```
Perhaps it would be good to test other commands such as delete too?

mingyi456 replied:
`find()` would also be good to test I guess?

ramapriyan912001 reviewed (commented):
Overall looks very good!
Perhaps the only minor change I might suggest would be to incorporate an empty line space between the declaration of fields and the declaration of the first constructor.

zzybluebell committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

zzybluebell committed:
A Gradle change the JDK to 11

mingyi456 commented on data/duke.txt:
```diff
@@ -0,0 +1,5 @@
+T | 1 | read book
+D | 0 | return book | 2021-08-12 15:30
+E | 0 | project meeting | 2021-08-22 21:30
+T | 1 | join sports club
+T | 0 | work
```
Not sure why there seems to be 2 data files. Maybe one of them should be deleted?

mingyi456 commented on src/main/java/Duke/Exceptions/DeleteWrongIndexException.java:
```diff
@@ -0,0 +1,14 @@
+public class DeleteWrongIndexException extends Exception {
+    /**
+     * The Constructor for DeleteWrongIndexException
+     * @param message
```
I think there needs to be a description for the parameter `message` in this JavaDocs line. This appears to be repeated many times, especially in all your `Exception` classes.

mingyi456 commented on src/main/java/Duke/Exceptions/EmptyTaskListException.java:
```diff
@@ -0,0 +1,14 @@
+public class EmptyTaskListException extends Exception{
+    /**
+     * The Constructor for EmptyTaskListException
+     * @param message
```
Please see previous comment regarding JavaDocs.

mingyi456 commented on src/main/java/Duke/Tasks/Task.java:
```diff
@@ -0,0 +1,82 @@
+
+
+/**
+ * Represents the Task class
```
It would be good to provide more descriptive details in this JavaDocs line, so that reviewers can understand this class better. Currently, I think this is rather vague.

mingyi456 commented on src/main/java/Duke/Tool/Parser.java:
```diff
@@ -0,0 +1,259 @@
+                        return new Event(subString_event.split(" /at ")[0],
+                                LocalDateTime.parse(subString_event.split(" /at ")[1], timeFormat), false);
+                    } catch (Exception e) {
+                        throw new WrongTimeFormatException(subString_event.split(" /at ")[1]);
+                    }
+
+                }
+
+            default:
+                throw new NoCommandException(instruction.name());
+            }
+        }
+
+        else {
```
I feel there are too many empty lines in this chunk of code, which affects readibility as a lot more scrolling is needed. Consider cutting down some of them? I noticed this pattern happening elsewhere, but this part is much more pronounced, so I believe you might be adhering to your own internal convention regarding how many blank lines to leave. Even so, I think breaking this convention is worth it if there are going to be this many empty lines.

mingyi456 commented on src/main/java/Duke/Tool/Parser.java:
```diff
@@ -0,0 +1,259 @@
+            return null;
+        }
+    }
+
+
+
```
Again, I feel there are too many blank lines here.

mingyi456 reviewed (commented):
Overall, the code quality is very much decent, with a few minor nits. I see almost every JavaDocs is too skeletonised, with almost no description of parameters if any. It might be better to have fewer, but more developed JavaDocs, and this would also fit the module requirements better. Also, I think the number of empty lines can be cut down.

zzybluebell committed:
The process of GUI

zzybluebell committed:
GUI successful UI

zzybluebell committed:
Gui Implementation

zzybluebell committed:
Merge branch 'GUI'

zzybluebell committed:
Fix the GUI minHeight label

zzybluebell committed:
A-Assertions: use assert features to detect various problems when enter todo, event and Deadline

zzybluebell committed:
Merge pull request #1 from zzybluebell/A-Assertions

A-Assertions

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
A-Streams, using stream to list down all the task in tasklist

zzybluebell committed:
Merge pull request #2 from zzybluebell/A-Streams

A streams

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
A-CodeQuality: check all codes

Javadoc: modified all in-apporiate Javadoc
eg. The Constructor -> Constructs

zzybluebell committed:
Merge pull request #3 from zzybluebell/branch-A-CodeQuality

Branch a code quality

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
B-DoWithinPeriodTasks:
Period extension: Add a new Period management in format "yyyy-MM-dd HH:mm" and displayed in "MMM dd yyyy HH:mm" format

zzybluebell committed:
Merge pull request #4 from zzybluebell/BCD-Extension

Bcd extension

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
Error detect:
For user enter wrong command or empty command

zzybluebell committed:
Merge pull request #6 from zzybluebell/ErrorInput

Error detect

zzybluebell committed:
A Better GUI

zzybluebell committed:
Merge branch 'ErrorInput'

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
Code Quality:
fix the file name Duke to duke

zzybluebell committed:
Readme.md

zzybluebell committed:
Merge pull request #7 from zzybluebell/A-BetterGui

A better gui

zzybluebell committed:
Merge branch 'master' of https://github.com/zzybluebell/ip

zzybluebell committed:
Ui image replaced

zzybluebell committed:
The code quality update
